franklin/github/views.py

```diff
@@ -111,10 +122,13 @@ def register_repo(request):
                     return Response(status=response.status_code)
                 else:
                     return Response(status=status.HTTP_201_CREATED)
-            else:
-                message = 'Current user does not have deployment permissions'
-                logger.warn(message + ' | %s | %s', request.user, site)
-                return Response(message, status=status.HTTP_403_FORBIDDEN)
+            if request.method == 'DELETE':
+                site.delete()
+                return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
piggybacking on the existing `/registration` endpoint gets us a lot for free here. Plus one for being RESTful.